src/paper/views/paper_views.py

```diff
@@ -1086,6 +1097,90 @@ def edit_file_extract(self, request, pk=None):
         )
         return Response(status=status.HTTP_200_OK)
 
+    @action(
+        detail=False,
+        methods=["get"],
+        permission_classes=[AllowAny],
+    )
+    def retrieve_by_doi(self, request):
```

Comment:
This will synchronously process paper from openalex if not in our database. 
LMK if anyone has better ideas of how to do this